drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/InvocationResponseFuture.java

```diff
@@ -78,7 +83,21 @@ private synchronized void tryConnect()
                 try {
                     if (channelFuture.isSuccess()) {
                         // Netty future listener generic type declaration requires a cast when used with a lambda
-                        tryInvocation((Channel) channelFuture.getNow());
+                        Channel channel = (Channel) channelFuture.getNow();
+                        ThrottleLock throttleLock = channel.attr(THROTTLE_LOCK_KEY).get();
+                        //If the connection is pooled, check for throttling else do direct invocation
+                        if (throttleLock != null) {
+                            synchronized (throttleLock) {
+                                while (!channel.isWritable()) {
+                                    log.info("Waiting since channel " + channel.id().asShortText() + " is not writeable");
+                                    throttleLock.wait(TimeUnit.MILLISECONDS.toMillis(100));
```

Comment:
I think this is safe since 
1. We are acquiring the object lock first.
2. Waiting for only 100 ms
3. Adding a while loop to check for spurious wakeups. 
Let me know if you are seeing any issues with this approach.